packages/compass-databases-navigation/src/database-item.tsx

```diff
@@ -26,6 +26,7 @@ const buttonReset = css({
 });
 
 const expandButton = css({
+  display: 'flex',
```

Comment:
Fine as-is, but just FYI setting font-size to zero (like the previous component did) would be a more direct fix to the alignment issue that you are seeing. Changing the parent display type to flex fixes it, but as a side-effect of converting children from inline elements (which svg-s are by default) to flex children